### PR TITLE
Prune device logs task and add server date column

### PR DESCRIPTION
**Pull request by snopoke**
http://manage.dimagi.com/default.asp?170335

@benrudolph need to delete old data in the table before merging

cc @kaapstorm

### CODE REVIEW

**Comment by benrudolph** [comment]
nice. why do we need to delete the old data first? looks like the celery task depends on a field that was already present

File: corehq/ex-submodules/phonelog/models.py [hunk 1]
```diff
@@ -10,6 +10,7 @@ class DeviceReportEntry(models.Model):
     msg = models.TextField()
     type = models.CharField(max_length=32, db_index=True)
     date = models.DateTimeField(db_index=True)
+    server_date = models.DateTimeField(null=True, db_index=True)
```
Comment (benrudolph) [inline corehq/ex-submodules/phonelog/models.py]: is this just a nice to have? don't see it actually used anywhere

**Comment by benrudolph** [comment]
oops disregard comments, just read the ticket. should really read that before reviewing

**Comment by czue** [comment]
@snopoke what's the status of this?

**Comment by snopoke** [comment]
I have a script to run that will drop all the indexes and delete all the
old data but have been cautious to run it since the db IO is already quite
high. Also no idea how long it will take.

**Comment by snopoke** [comment]
@czue this is GTG

**Comment by czue** [comment]
just to confirm - you think this is fine to run a normal migration as part of the normal deploy? or did we want to try and run it in a preindex directory in case it takes a while?

**Comment by snopoke** [comment]
No harm in going the preindex route but I don't expect it to take longer
than 10 mins.

**Comment by czue** [comment]
@proteusvacuum @dannyroberts @TylerSheffels feel free to merge this just before deploy. note that the migration may run for up to 10 minutes so could be worth doing it manually in the preindex directory. not merging in case you want to coordinate / plan specifically.

**Comment by proteusvacuum** [comment]
There probably won't be a deploy today because there is a big reindex happening. 
Pinging @esoergel who I think is on deploy tomorrow.

**Comment by snopoke** [comment]
@esoergel just making sure this gets in today

**Comment by esoergel** [comment]
@snopoke I'll preindex separately.

**Comment by esoergel** [comment]
FYI, the migration ran just fine.  Doing the normal deploy now.

**Pull request merged**